src/@utils/provider.ts

```diff
@@ -15,44 +14,44 @@ export async function getEncryptedFileUrls(
   accountId: string
 ): Promise<string> {
   try {
-    // https://github.com/oceanprotocol/provider/blob/v4main/API.md#encrypt-endpoint
-    const url = `${providerUrl}/api/v1/services/encrypt`
-    const response: AxiosResponse<{ encryptedDocument: string }> =
-      await axios.post(url, {
-        documentId: did,
-        signature: '', // TODO: add signature
-        publisherAddress: accountId,
-        document: files
-      })
-    return response?.data?.encryptedDocument
+    const response = await ProviderInstance.encrypt(
+      did,
+      accountId,
+      files,
+      providerUrl,
+      fetch
+    )
+    if (!response) return
+    return response
   } catch (error) {
     console.error('Error parsing json: ' + error.message)
   }
 }
 
-export async function getFileInfo(
-  url: string | DID,
-  providerUrl: string,
-  cancelToken: CancelToken
-): Promise<FileMetadata[]> {
-  let postBody
-  try {
-    if (url instanceof DID) postBody = { did: url.getDid() }
-    else postBody = { url }
+async function fetchMethod(provider: string, url: string): Promise<Response> {
```

Comment:
No need to create a separate function, used like this https://github.com/oceanprotocol/market/blob/16475b55d13b474840a6567e346ab3e466f37b7b/src/components/Publish/index.tsx#L188